### PR TITLE
location safety on excel reports

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?260587#1383161

@mkangia @esoergel

### CODE REVIEW

**Comment by esoergel** [comment]
@sravfeyn is my understanding correct?  This will verify that the report class is accessible, but it won't re-verify the report, right?  If user A generated the report, then sent a link to user B, user B would be able to export user A's version of the report, even if their permissions were different.  Is this a URL that's sent in email or just something used internally through javascript (like, you click the "download" button after it's finished generating, and that makes this request).  I think this is probably fine, and checking the locations again would require passing those as context to redis, which might not be straightforward.

@sheelio thoughts on this?

**Comment by sheelio** [comment]
@esoergel 

Its a bit of a weird situation, but I would think that the act of a user emailing a report would essentially "approve" the recipient for its contents (so even if the recipient didn't have permissions, the sender provided them with permission).

**Comment by sravfeyn** [comment]
Ethan, yeah, you are right. I thought about the same issue a bit, but this felt like still acceptable.

File: corehq/apps/reports/views.py [hunk 1]
```diff
@@ -2008,6 +2010,16 @@ def mk_date_range(start=None, end=None, ago=timedelta(days=7), iso=False):
...
+def _is_location_safe_report_class(request, domain, export_hash, format):
+    cache = get_redis_client()
+
+    content = cache.get(export_hash)
+    if content is not None:
+        report_class, report_file = content
+        return report_class_is_location_safe(report_class)
+
+
+@conditionally_location_safe(_is_location_safe_report_class)
```
Comment (mkangia) [inline corehq/apps/reports/views.py]: Probably does not make a difference, just the decorator order would look better if this comes after all conditions are satisfied
```
@login_and_domain_required
@require_GET
@conditionally_location_safe(_is_location_safe_report_class)
```

Unless there is a reason for putting it at the top instead.

**Review by mkangia — APPROVED** [review]

**Review by esoergel — APPROVED** [review]

**Pull request merged**